### PR TITLE
add DAG json to result folder

### PR DESCRIPTION
**Pull request by cdebourcy**
image

image

### CODE REVIEW

File: app/assets/src/components/ResultsFolder.jsx [hunk 1]
```diff
@@ -1,12 +1,123 @@
...
+
+class ConfigFile extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.stageDagJson = props.stageDagJson;
+  }
+
+  download_string2file(str) {
+    let file = new Blob([str], { type: "text/plain" });
+    let download_url = URL.createObjectURL(file);
```
Comment (cdebourcy) [inline app/assets/src/components/ResultsFolder.jsx]: @MarkAZhang are there any risks to this?

File: app/assets/src/components/ResultsFolder.jsx [hunk 2]
```diff
@@ -1,12 +1,123 @@
...
+    if (url) {
+      location.href = `${url}`;
+    }
+  }
+
+  render() {
+    return (
+      <tr
+        className={`${this.file.url ? "" : "disabled-"}file-link`}
+        onClick={this.download.bind(this, this.file.url)}
```
Comment (jshoe) [inline app/assets/src/components/ResultsFolder.jsx]: For standardization this should use utils/links.js

File: app/assets/src/components/ResultsFolder.jsx [hunk 3]
```diff
@@ -1,12 +1,123 @@
...
+  download(url) {
+    if (url) {
+      location.href = `${url}`;
+    }
+  }
+
+  render() {
+    return (
+      <tr
+        className={`${this.file.url ? "" : "disabled-"}file-link`}
```
Comment (jshoe) [inline app/assets/src/components/ResultsFolder.jsx]: re className all new code should use CSS Modules and camelCase. See ui/controls/Tabs.jsx and tabs.scss

File: app/assets/src/components/ResultsFolder.jsx [hunk 4]
```diff
@@ -1,12 +1,123 @@
...
+  }
+}
+
+class ConfigFile extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.stageDagJson = props.stageDagJson;
+  }
+
+  download_string2file(str) {
```
Comment (jshoe) [inline app/assets/src/components/ResultsFolder.jsx]: same with these functions for camelCase and CSS Modules

Comment (jshoe) [reply app/assets/src/components/ResultsFolder.jsx]: well, I guess the CSS is not really new

File: app/models/pipeline_run_stage.rb [hunk 5]
```diff
@@ -58,6 +58,11 @@ def checked?
     job_status == STATUS_CHECKED
   end
 
+  def redacted_dag_json
+    # redact any s3 paths
+    dag_json.gsub(%r{(\"s3://).*(\")}, '"s3://..."')
```
Comment (yunfangjuan) [inline app/models/pipeline_run_stage.rb]: awesome!

File: app/assets/src/components/ResultsFolder.jsx (hunk 4 above)
Comment (cdebourcy) [reply app/assets/src/components/ResultsFolder.jsx]: yeah the CSS is not new and the download function is also not new.  Just broke single ResultsFolder component into some smaller pieces.
I will rename `downloadString2file`. This whole page is going to be redesigned and refactored in the future

File: app/assets/src/components/ResultsFolder.jsx (hunk 2 above)
Comment (cdebourcy) [reply app/assets/src/components/ResultsFolder.jsx]: ok moved

File: app/assets/src/components/ResultsFolder.jsx (hunk 3 above)
Comment (cdebourcy) [reply app/assets/src/components/ResultsFolder.jsx]: I couldn't get the modules to work. Will bear in mind for new CSS in the future and seek clarification

**Pull request merged**